### PR TITLE
Changed the working of UpVotes

### PR DESCRIPTION
**Pull request by sshivam95**

### CODE REVIEW

File: venter/views.py [hunk 1]
```diff
@@ -7,14 +7,16 @@
...
+from venter.models import TagUris
 
 from venter.serializers import ComplaintSerializer
 from venter.serializers import ComplaintPostSerializer
 from venter.serializers import CommentPostSerializer
 from venter.serializers import CommentSerializer
 
+
 class ComplaintViewSet(viewsets.ModelViewSet):
+    count = 0
```
Comment (pulsejet) [inline venter/views.py]: Can't see how these changes are necessary

File: venter/tests.py [hunk 2]
```diff
@@ -66,10 +66,24 @@ def test_complaint(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.data['description'], 'test')
 
-        response = self.client.put(url, data, format='json')
-        self.assertEqual(response.status_code, 200)
+        url = '/api/venter/complaints/'+cid+'/upvote'
```
Comment (pulsejet) [inline venter/tests.py]: Add spacing before and after `+`. A cleaner way would be to use the `%` operator

File: venter/tests.py [hunk 3]
```diff
@@ -66,10 +66,24 @@ def test_complaint(self):
...
+
+        url += '?action='
```
Comment (pulsejet) [inline venter/tests.py:75]: Put this in the definition `url` itself

File: venter/views.py [hunk 4]
```diff
@@ -77,17 +79,33 @@ def create(cls, request):
             Complaints.objects.get(id=complaint.id)
         ).data, status=201)
 
-    def update(self, request, pk):
+    @login_required_ajax
+    def upVote(self, request, pk):
+        """UpVote or unUpVote a complaint {?action}=0,1"""
```
Comment (pulsejet) [inline venter/views.py]: "Upvote or un-upvote";

File: venter/views.py [hunk 5]
```diff
@@ -77,17 +79,33 @@ def create(cls, request):
...
         return Response(ComplaintSerializer(
             Complaints.objects.get(id=complaint.id)
-        ).data)
+        ).data, status=204)
 
     def get_complaint(self, pk):
         """Shortcut for get_object_or_404 with pk"""
         return get_object_or_404(self.queryset, id=pk)
 
+
```
Comment (pulsejet) [inline venter/views.py]: Unnecessary whitespace change

File: venter/views.py [hunk 6]
```diff
@@ -77,17 +79,33 @@ def create(cls, request):
...
+            complaint.users_up_voted.remove(self.request.user.profile)
+        elif value == "1":
+            complaint.users_up_voted.add(self.request.user.profile)
+        else:
+            return Response({"message": "Invalid Action"}, status=400)
+
         return Response(ComplaintSerializer(
             Complaints.objects.get(id=complaint.id)
-        ).data)
+        ).data, status=204)
```
Comment (pulsejet) [inline venter/views.py]: `204` should not have a body. Either remove the body or change the status code to `200`. This is plain _wrong_ the way it is right now.

File: venter/views.py [hunk 7]
```diff
@@ -77,17 +79,33 @@ def create(cls, request):
...
+        """UpVote or unUpVote a complaint {?action}=0,1"""
+
         complaint = self.get_complaint(pk)
-        complaint.users_up_voted.add(self.request.user.profile)
+
+        value = request.GET.get("action")
+        if value is None:
+            return Response({"message": "{?action} is required"}, status=400)
+
+            # Check possible actions
```
Comment (pulsejet) [inline venter/views.py]: Why is this comment indented?

File: venter/views.py [hunk 8]
```diff
@@ -77,17 +79,33 @@ def create(cls, request):
             Complaints.objects.get(id=complaint.id)
         ).data, status=201)
 
-    def update(self, request, pk):
+    @login_required_ajax
+    def upVote(self, request, pk):
```
Comment (pulsejet) [inline venter/views.py]: Use snake_case for function names

**Review by pulsejet — CHANGES_REQUESTED** [review]

File: venter/tests.py [hunk 9]
```diff
@@ -1,7 +1,14 @@
+"""
+To run the tests fire this command:
+
+coverage run manage.py test venter --settings backend.settings_test && coverage html
```
Comment (pulsejet) [inline venter/tests.py]: No meta information in source code please. If this command changes later, this change will stay and confuse the next person. Instead, make a separate PR to add this to README.

File: venter/tests.py [hunk 10]
```diff
@@ -1,7 +1,14 @@
...
+"""
+
 from rest_framework.test import APITestCase
 from login.tests import get_new_user
 from venter.models import Complaints, TagUris, Comment, ComplaintMedia
 
+
```
Comment (pulsejet) [inline venter/tests.py]: Unnecessary whitespace change

File: venter/tests.py [hunk 11]
```diff
@@ -34,7 +41,7 @@ def test_complaint(self):
         TagUris.objects.create(tag_uri='garbage')
         data = {
             'description': 'test',
-            'tags':['flexes', 'garbage'],
+            'tags': ['flexes', 'garbage'],
```
Comment (pulsejet) [inline venter/tests.py:37]: This is fine for now, but *please* keep refactor and feature PRs separate.

File: venter/views.py [hunk 12]
```diff
@@ -7,13 +7,14 @@
...
 from venter.models import ComplaintMedia
-from venter.models import  TagUris
+from venter.models import TagUris
 
 from venter.serializers import ComplaintSerializer
 from venter.serializers import ComplaintPostSerializer
 from venter.serializers import CommentPostSerializer
 from venter.serializers import CommentSerializer
 
+
```
Comment (pulsejet) [inline venter/views.py]: Whitespace change. You can view these on your command prompt before you commit by `git diff`

File: venter/views.py [hunk 13]
```diff
@@ -77,17 +78,33 @@ def create(cls, request):
...
+            complaint.users_up_voted.remove(self.request.user.profile)
+        elif value == "1":
+            complaint.users_up_voted.add(self.request.user.profile)
+        else:
+            return Response({"message": "Invalid Action"}, status=400)
+
         return Response(ComplaintSerializer(
             Complaints.objects.get(id=complaint.id)
-        ).data)
+        ).data, status=200)
```
Comment (pulsejet) [inline venter/views.py:100]: I'll let this pass, but you don't really need to specify this explicitly, since the default status code is `200` anyway.

Comment (sshivam95) [reply venter/views.py:100]: So should I change?

Comment (pulsejet) [reply venter/views.py:100]: Your call. Fix the couple of whitespace changes and the `url` definition thing, and this should be fine for a merge.

File: venter/tests.py (hunk 3 above)
Comment (pulsejet) [reply venter/tests.py:75]: This makes more sense now

File: venter/views.py (hunk 13 above)
Comment (pulsejet) [reply venter/views.py:100]: *just the couple of whitespace changes.

Comment (pulsejet) [reply venter/views.py:100]: Yes, I already marked that resolved.

Comment (sshivam95) [reply venter/views.py:100]: Check the tests.py file once again

Comment (pulsejet) [reply venter/views.py:100]: That's what I meant, nevermind

File: venter/views.py (hunk 5 above)
Comment (sshivam95) [reply venter/views.py]: Changed the whitespace

**Review by pulsejet — APPROVED** [review]

**Comment by sshivam95** [comment]
Thanks
